jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -218,112 +195,85 @@ typedef struct
-                                                   ecma_property_t *prop_p,
-                                                   ecma_value_t value);
-
-extern ecma_object_t* ecma_get_named_accessor_property_getter (const ecma_property_t *prop_p);
-extern ecma_object_t* ecma_get_named_accessor_property_setter (const ecma_property_t *prop_p);
-extern void ecma_set_named_accessor_property_getter (ecma_object_t* object_p,
-                                                     ecma_property_t *prop_p,
-                                                     ecma_object_t *getter_p);
-extern void ecma_set_named_accessor_property_setter (ecma_object_t* object_p,
-                                                     ecma_property_t *prop_p,
-                                                     ecma_object_t *setter_p);
-extern bool ecma_is_property_writable (ecma_property_t* prop_p);
-extern void ecma_set_property_writable_attr (ecma_property_t* prop_p, bool is_writable);
-extern bool ecma_is_property_enumerable (ecma_property_t* prop_p);
-extern void ecma_set_property_enumerable_attr (ecma_property_t* prop_p, bool is_enumerable);
-extern bool ecma_is_property_configurable (ecma_property_t* prop_p);
-extern void ecma_set_property_configurable_attr (ecma_property_t* prop_p, bool is_configurable);
-
-extern bool ecma_is_property_lcached (ecma_property_t *prop_p);
-extern void ecma_set_property_lcached (ecma_property_t *prop_p,
-                                       bool is_lcached);
+extern ecma_object_t* ecma_create_object (ecma_object_t *, bool, ecma_object_type_t);
+extern ecma_object_t* ecma_create_decl_lex_env (ecma_object_t *);
+extern ecma_object_t* ecma_create_object_lex_env (ecma_object_t *, ecma_object_t *, bool);
+extern bool __attr_pure___ ecma_is_lexical_environment (const ecma_object_t *);
+extern bool __attr_pure___ ecma_get_object_extensible (const ecma_object_t *);
+extern void ecma_set_object_extensible (ecma_object_t *, bool);
+extern ecma_object_type_t __attr_pure___ ecma_get_object_type (const ecma_object_t *);
+extern void ecma_set_object_type (ecma_object_t *, ecma_object_type_t);
+extern ecma_object_t* __attr_pure___ ecma_get_object_prototype (const ecma_object_t *);
+extern bool __attr_pure___ ecma_get_object_is_builtin (const ecma_object_t *);
+extern void ecma_set_object_is_builtin (ecma_object_t *, bool);
+extern ecma_lexical_environment_type_t __attr_pure___ ecma_get_lex_env_type (const ecma_object_t *);
+extern ecma_object_t* __attr_pure___ ecma_get_lex_env_outer_reference (const ecma_object_t *);
+extern ecma_property_t* __attr_pure___ ecma_get_property_list (const ecma_object_t *);
+extern ecma_object_t* __attr_pure___ ecma_get_lex_env_binding_object (const ecma_object_t *);
+extern bool __attr_pure___ ecma_get_lex_env_provide_this (const ecma_object_t *);
+
+extern ecma_property_t* ecma_create_internal_property (ecma_object_t *, ecma_internal_property_id_t);
```

Comment:
BTW: I've noticed that there are a lot of function return values which incorrect format as before the `*` character  there is no space. Do we want to fix those also? or maybe later?